src/Microsoft.Kusto.ServiceLayer/DataSource/Intellisense/KustoIntellisenseClient.cs

```diff
@@ -42,16 +42,10 @@ private GlobalState LoadSchemaState(string databaseName, string clusterName)
             if (!string.IsNullOrWhiteSpace(databaseName))
             {
                 var source = new CancellationTokenSource();
-                Parallel.Invoke(() =>
-                    {
-                        tableSchemas =
-                            _kustoClient.ExecuteQueryAsync<ShowDatabaseSchemaResult>($".show database {databaseName} schema", source.Token, databaseName)
-                                .Result;
-                    },
-                    () =>
-                    {
-                        functionSchemas = _kustoClient.ExecuteQueryAsync<ShowFunctionsResult>(".show functions", source.Token, databaseName).Result;
-                    });
+                var tableQuery = $".show database {KustoQueryUtils.EscapeName(databaseName)} schema";
+                
+                tableSchemas = _kustoClient.ExecuteQueryAsync<ShowDatabaseSchemaResult>(tableQuery, source.Token, databaseName).Result;
+                functionSchemas = _kustoClient.ExecuteQueryAsync<ShowFunctionsResult>(".show functions", source.Token, databaseName).Result;
```

Comment:
Removed Parallel.Invoke since both threads are acting against ExecuteQueryAsync which is a single instance and is still being executed with the thread blocked.

Reply:
Even though the ICslQueryProvider.ExecuteQuery is synchronous, it might be issuing the queries in parallel underneath. We have another parallel execution in the KustoClient class, and I remember that made a difference when we issued multiple parallel queries. Better to be parallel on our side for perf reasons, unless we are calling code which is not thread safe.